GPSd support

mckaymatthew opened the pull request:
This pull request partially address #909. I understand the desire for a more flexible approach for alternative data sources as discussed in the issue, but this covers many use cases and is fairly reasonable. It is also fairly loosely integrated and doesn't lock future work. 

Please take a look and provide feedback. 

When compiled, and when enabled, messages received have the GPS Status, Fix Mode, and LLA appended to it. 

A couple of notes I have:
- Optionally compiled support. Takes the "AUTO" approach SoapySDR support uses. 
- Command line argument, -u requires a "hostname:port" argument. 
- Failure cases considered:
  - Startup behavior is to exit if gpsd connection is not successful. 
  - Runtime behavior is to auto-reconnect if gpsd connection was lost. This does print to stderr at runtime.
- Builds on all my platforms, all tests pass
  - Centos 7. x86_64. 1663 records tested, 0 have failed
  - Ubuntu 18.04.3 LTS. x86_64. 1663 records tested, 0 have failed
  - Raspbian GNU/Linux 9. armv7l. 1663 records tested, 0 have failed

zuckschwerdt commented:
Thanks. It'll take some time for me to review, lots of backlog.
One first note: if not for legacy reasons CMake switches should not be a magic AUTO. It is reasonable to require people to set a YES and then break if the lib isn't found.

merbanan commented on src/rtl_433.c:
```diff
@@ -1054,6 +1063,35 @@ static void parse_conf_option(r_cfg_t *cfg, int opt, char *arg)
+            if(count != 1) {
+                (void)fprintf(stderr,"GPS requires a host and port.\n");
+                usage(1);
+            }
```
How about to default to localhost and port 2947 if no arguments are given?

gdt replied:
I agree that defaulting to localhost:2947 is good.   My guess at the normal use case is a laptop that is moving, so having gpsd on the same laptop is the obvious approach.

alphafox02 replied:
@merbanan it doesn’t seem like this was ever added to main? Was there not enough interest or need?

alphafox02 replied:
Actually I didn’t read closely enough 
https://github.com/merbanan/rtl_433/pull/1636

merbanan commented:
One more comment is the use of ifdefs. I really dislike them in common code but I'm not sure it is worth the complexity to get rid of them. Anyway the implementation looks clean.

mckaymatthew commented:
Thanks for the quick feedback. 
- I made it a "NO" option by default.
- I added a small section to BUILDING.md mentioning the option.
- I agree ifdefs are undesirable. The other option I considered was runtime dynamic linking but this has its own problems. Sounds like ifdef is fine for now.
- Made -u have an optional argument. I've never used getopt before... this proved to be confusing to implement. 
- Fixed stylecheck error.
